x-pack/plugins/stack_alerts/public/alert_types/es_query/expression/use_test_query.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useState, useCallback } from 'react';
+import { i18n } from '@kbn/i18n';
+import * as estypes from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+
+/**
+ * Hook used to test the data fetching execution by returning a number of found documents
+ * Or in error in case it's failing
+ */
+export function useTestQuery(fetch: () => Promise<{ nrOfDocs: number; timeWindow: string }>) {
+  const [testQueryResult, setTestQueryResult] = useState<string | null>(null);
+  const [testQueryError, setTestQueryError] = useState<string | null>(null);
+
+  const onTestQuery = useCallback(async () => {
+    setTestQueryError(null);
+    setTestQueryResult(null);
+    try {
+      const { nrOfDocs, timeWindow } = await fetch();
+
+      setTestQueryResult(
+        i18n.translate('xpack.stackAlerts.esQuery.ui.numQueryMatchesText', {
+          defaultMessage: 'Query matched {count} documents in the last {window}.',
+          values: { count: nrOfDocs, window: timeWindow },
+        })
+      );
+    } catch (err) {
+      const message = err?.body?.attributes?.error?.root_cause[0]?.reason || err?.body?.message;
+      setTestQueryError(
```

Comment:
Is that possible `err?.body?.message` will be `undefined`?

Reply:
it should not be, but in this case I've moved code from the recent implementation